From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Phoenix616 <mail@moep.tv>
Date: Tue, 7 Jul 2020 00:22:11 +0100
Subject: [PATCH] Fix exp drop of zombie pigmen (MC-56653)


diff --git a/src/main/java/de/minebench/origami/OrigamiConfig.java b/src/main/java/de/minebench/origami/OrigamiConfig.java
index c0d6c1eb7ddec5ae118a1a9fc6c00c5b0b9b0a0c..f96a045ab9902599c3c1e76bb6810807eab36602 100644
--- a/src/main/java/de/minebench/origami/OrigamiConfig.java
+++ b/src/main/java/de/minebench/origami/OrigamiConfig.java
@@ -132,5 +132,9 @@ public final class OrigamiConfig {
         private void observerClock() {
             disableObserverClocks = getBoolean("disable-observer-clocks", disableObserverClocks);
         }
+        public boolean pigmenDontTargetUnlessHit = false;
+        private void pigmenDontTargetUnlessHit() {
+            pigmenDontTargetUnlessHit = getBoolean("pigmen.dont-target-unless-hit", pigmenDontTargetUnlessHit);
+        }
     }
 }
diff --git a/src/main/java/net/minecraft/world/entity/monster/EntityPigZombie.java b/src/main/java/net/minecraft/world/entity/monster/EntityPigZombie.java
index d10d1b768601236b9892461ee41d61c7239d1a07..1331b7525be64ecf5115dcafc6511837eff5b7ec 100644
--- a/src/main/java/net/minecraft/world/entity/monster/EntityPigZombie.java
+++ b/src/main/java/net/minecraft/world/entity/monster/EntityPigZombie.java
@@ -88,7 +88,7 @@ public class EntityPigZombie extends EntityZombie implements IEntityAngerable {
     protected void mobTick() {
         AttributeModifiable attributemodifiable = this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED);
 
-        if (this.isAngry()) {
+        if (this.isAngry() && !this.world.origamiConfig.pigmenDontTargetUnlessHit) { // Origami - fix MC-56653
             if (!this.isBaby() && !attributemodifiable.a(EntityPigZombie.c)) {
                 attributemodifiable.b(EntityPigZombie.c);
             }
