From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DoctaEnkoda <bierquejason@gmail.com>
Date: Sun, 18 Jul 2021 12:28:04 +0200
Subject: [PATCH] Fix Issue PaperMC#6028


diff --git a/src/main/java/net/minecraft/world/level/levelgen/feature/TreeFeature.java b/src/main/java/net/minecraft/world/level/levelgen/feature/TreeFeature.java
index a95650a5e69897441d56e829968d62566406755c..124cc1edfebb59f1442653270f4c93c25ccffb27 100644
--- a/src/main/java/net/minecraft/world/level/levelgen/feature/TreeFeature.java
+++ b/src/main/java/net/minecraft/world/level/levelgen/feature/TreeFeature.java
@@ -143,7 +143,7 @@ public class TreeFeature extends Feature<TreeConfiguration> {
             worldGenLevel.setBlock(pos, state, 19);
         };
         boolean bl = this.doPlace(worldGenLevel, random, blockPos, biConsumer, biConsumer2, treeConfiguration);
-        if (bl && (!set.isEmpty() || !set2.isEmpty())) {
+        if (bl && !set.isEmpty() && !set2.isEmpty()) { // PaperMC - Fix PaperMC#6028
             if (!treeConfiguration.decorators.isEmpty()) {
                 List<BlockPos> list = Lists.newArrayList(set);
                 List<BlockPos> list2 = Lists.newArrayList(set2);
