From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Thu, 9 Jun 2016 00:57:38 -0400
Subject: [PATCH] Don't use force unload for keep spawn setting change


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index d4ed84e5825e5fb544980a20f0ac4eec48ddd799..460aa284fa329dd4065444437446c793156a564b 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -513,7 +513,7 @@ public class CraftWorld implements World {
 
     @Override
     public boolean unloadChunk(int x, int z) {
-        return unloadChunk(x, z, true);
+        return unloadChunkRequest(x, z); // Paper
     }
 
     @Override
