From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kyle Wood <kyle@denwav.dev>
Date: Wed, 16 Jun 2021 01:55:05 -0500
Subject: [PATCH] Build changes


diff --git a/build.gradle.kts b/build.gradle.kts
index c90209011a266e71511954f23a1e42efa8c62a54..4a6a7db602292a89c3675bd1c9aa8efe0fe8d856 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -14,7 +14,7 @@ repositories {
 }
 
 dependencies {
-    implementation(project(":paper-api"))
+    implementation(project(":patina-api")) // Patina
     api("com.mojang:brigadier:1.0.18")
 
     compileOnly("it.unimi.dsi:fastutil:8.5.6")
