From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kyle Wood <kyle@denwav.dev>
Date: Wed, 16 Jun 2021 01:55:05 -0500
Subject: [PATCH] Build changes


diff --git a/build.gradle.kts b/build.gradle.kts
index f7bf0ff771e6f951f0f1d97b9e3b3fa4cd9dc5c2..d4b1533cb0d289d413194fb6fc5e0012b4f17887 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -16,7 +16,7 @@ repositories {
 }
 
 dependencies {
-    implementation(project(":Paper-API"))
+    implementation(project(":Patina-API")) // Patina
     api("com.mojang:brigadier:1.0.18")
 
     compileOnly("it.unimi.dsi:fastutil:8.2.2")
