From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Mon, 28 Jun 2021 10:36:43 +0000
Subject: [PATCH] Purpur dependencies


diff --git a/build.gradle.kts b/build.gradle.kts
index 07945418e95cbe07c45bfa815d828fa311c7f033..1d979718c8c2db437c03945a304098a5a76d9952 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -50,6 +50,7 @@ dependencies {
     apiAndDocs("net.kyori:adventure-text-serializer-legacy")
     apiAndDocs("net.kyori:adventure-text-serializer-plain")
     apiAndDocs("net.kyori:adventure-text-logger-slf4j")
+    api("net.kyori:adventure-text-minimessage:4.1.0-SNAPSHOT") // Purpur
     api("org.apache.logging.log4j:log4j-api:$log4jVersion")
     api("org.slf4j:slf4j-api:$slf4jVersion")
     api("io.sentry:sentry:5.4.0") // Pufferfish
