From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Mon, 28 Jun 2021 10:36:43 +0000
Subject: [PATCH] Purpur dependencies


diff --git a/build.gradle.kts b/build.gradle.kts
index 3676035ba011fb874ad6bb8995940d9bb2a0f0ae..82dadb8bf06013022e7a4a2d9078fa38f784f10b 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -44,6 +44,7 @@ dependencies {
     apiAndDocs("net.kyori:adventure-text-serializer-legacy")
     apiAndDocs("net.kyori:adventure-text-serializer-plain")
     apiAndDocs("net.kyori:adventure-text-logger-slf4j")
+    api("net.kyori:adventure-text-minimessage:4.1.0-SNAPSHOT") // Purpur
     api("org.apache.logging.log4j:log4j-api:$log4jVersion")
     api("org.slf4j:slf4j-api:$slf4jVersion")
     api("io.sentry:sentry:5.4.0") // Pufferfish
