From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Mon, 28 Jun 2021 10:36:43 +0000
Subject: [PATCH] Purpur dependencies


diff --git a/build.gradle.kts b/build.gradle.kts
index b8fa29c4e39c4587855e6c8be3f5b373bd841ba3..2a698d9d00ee57aff52023b19605dd149792e22d 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -39,6 +39,7 @@ dependencies {
     apiAndDocs("net.kyori:adventure-text-serializer-legacy")
     apiAndDocs("net.kyori:adventure-text-serializer-plain")
     apiAndDocs("net.kyori:adventure-text-logger-slf4j")
+    api("net.kyori:adventure-text-minimessage:4.1.0-SNAPSHOT") // Purpur
     api("org.apache.logging.log4j:log4j-api:2.17.1")
     api("org.slf4j:slf4j-api:1.8.0-beta4")
     api("io.sentry:sentry:5.4.0") // Pufferfish
