From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Mon, 28 Jun 2021 10:36:43 +0000
Subject: [PATCH] Purpur dependencies


diff --git a/build.gradle.kts b/build.gradle.kts
index 0007c6b1d81373cb6592c8b9e02f464405050f68..38c1dd17f42e549c87c0573ea2c0b06614041152 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -39,6 +39,7 @@ dependencies {
     apiAndDocs("net.kyori:adventure-text-serializer-gson")
     apiAndDocs("net.kyori:adventure-text-serializer-legacy")
     apiAndDocs("net.kyori:adventure-text-serializer-plain")
+    api("net.kyori:adventure-text-minimessage:4.1.0-SNAPSHOT") // Purpur
     api("org.apache.logging.log4j:log4j-api:2.17.1")
     api("org.slf4j:slf4j-api:1.8.0-beta4")
     api("io.sentry:sentry:5.4.0") // Pufferfish
