From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: YatopiaMC <yatopiamc@gmail.com>
Date: Sun, 17 Jan 2021 15:37:52 -0600
Subject: [PATCH] Yatopia Config & Redirect Config


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 6dbf856ff422a02cd035d2a9a8535b97bc76ac2f..126727951ef627a97d4b283b9d6857e763dd546f 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1790,6 +1790,22 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Purpur end
 
+        // Origami start - add config to timings report
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getOrigamiConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Origami end
+
+        // Yatopia start - add config to timings report
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getYatopiaConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Yatopia end
+
         /**
          * Sends the component to the player
          *
