From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Mon, 2 Sep 2024 16:23:31 +1200
Subject: [PATCH] remove getServerName from purpur


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 8f3462f3242ec9f583a5d7a297fdd9c6f43cfc31..a5d77a6530b5a68ebf9cc725c57615064327b376 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2923,15 +2923,6 @@ public final class Bukkit {
     }
 
     // Purpur start
-    /**
-     * Get the name of this server
-     * @return the name of the server
-     */
-    @NotNull
-    public static String getServerName() {
-        return server.getServerName();
-    }
-
     /**
      * Check if server is lagging according to laggy threshold setting
      *
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 766b4908ba5057bcda566f8a8cf1914416ede910..e11c7ceb9b11c3b7dc5f6ebaa542b8d90d182110 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2568,13 +2568,6 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     // Paper end - Folia region threading API
 
     // Purpur start
-    /**
-     * Get the name of this server
-     * @return the name of the server
-     */
-    @NotNull
-    String getServerName();
-
     /**
      * Check if server is lagging according to laggy threshold setting
      *
