From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Toffikk <slodkitofik@gmail.com>
Date: Thu, 17 Jun 2021 14:44:11 +0200
Subject: [PATCH] Fix Javadoc generation


diff --git a/build.gradle.kts b/build.gradle.kts
index 206ac6d91e9935c12e65244fe5ed9c168781eb6b..9a756babb1ee54ab66f24a69ae47ca2f9bc7b03b 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -74,6 +74,8 @@ tasks.jar {
 }
 
 tasks.withType<Javadoc>().configureEach {
+    (options as StandardJavadocDocletOptions).encoding = "UTF-8"
+    (options as StandardJavadocDocletOptions).addBooleanOption("html5", true)
     (options as StandardJavadocDocletOptions).links(
         "https://guava.dev/releases/21.0/api/docs/",
         "https://javadoc.io/doc/org.yaml/snakeyaml/1.28/",
