From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Toffikk <slodkitofik@gmail.com>
Date: Thu, 17 Jun 2021 14:44:11 +0200
Subject: [PATCH] Fix Javadoc generation


diff --git a/build.gradle.kts b/build.gradle.kts
index 407bf7c09304c76ef1dbec43bb5d665f02ce4840..c675d536f6c55562ee8ab2a45fef170406cb9a0d 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -70,6 +70,8 @@ tasks.jar {
 }
 
 tasks.withType<Javadoc>().configureEach {
+    (options as StandardJavadocDocletOptions).encoding = "UTF-8"
+    (options as StandardJavadocDocletOptions).addBooleanOption("html5", true)
     (options as StandardJavadocDocletOptions).links(
         "https://guava.dev/releases/21.0/api/docs/",
         "https://javadoc.io/doc/org.yaml/snakeyaml/1.27/",
