From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Mon, 9 Sep 2024 08:54:32 +1200
Subject: [PATCH] rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 652ff54e7c50412503725d628bfe72ed03059790..e1421e6135075ba8f8f4694f2666c13298f75f58 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -19,6 +19,9 @@ public interface ServerBuildInfo {
      */
     Key BRAND_PAPER_ID = Key.key("papermc", "paper");
 
+    Key BRAND_PUFFERFISH_ID = Key.key("pufferfish", "pufferfish"); // Patina
+    Key BRAND_PATINA_ID = Key.key("patina", "patina"); // Patina
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
