From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Sun, 18 Apr 2021 09:57:59 +0000
Subject: [PATCH] Remove unnecessary `LogManager.getLogger()`


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index eb44489ae2304c7e3602ed39597d9f0babaad413..554ce27420880a3e866a3982043672c69414906a 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -159,7 +159,7 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, ne
         @Override
         public synchronized void setSeed(long seed) {
             if (locked) {
-                LogManager.getLogger().error("Ignoring setSeed on Entity.SHARED_RANDOM", new Throwable());
+                LOGGER.error("Ignoring setSeed on Entity.SHARED_RANDOM", new Throwable()); // Patina - Remove unnecessary `LogManager.getLogger()`
             } else {
                 super.setSeed(seed);
                 locked = true;
@@ -2192,7 +2192,7 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, ne
                 try {
                     spawnReason = org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.valueOf(spawnReasonName);
                 } catch (Exception ignored) {
-                    LogManager.getLogger().error("Unknown SpawnReason " + spawnReasonName + " for " + this);
+                    LOGGER.error("Unknown SpawnReason " + spawnReasonName + " for " + this); // Patina - Remove unnecessary `LogManager.getLogger()`
                 }
             }
             if (spawnReason == null) {
