From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Thu, 1 Jul 2021 12:32:56 +0000
Subject: [PATCH] relocate


diff --git a/build.gradle.kts b/build.gradle.kts
index eb57b5946990fef0201785725086068df3a96f0c..01ef911a5d1737a5c255ebad104a809886daca4d 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -107,6 +107,8 @@ relocation {
     relocate("org.bukkit.craftbukkit" to "org.bukkit.craftbukkit.v$craftbukkitPackageVersion") {
         exclude("org.bukkit.craftbukkit.Main*")
     }
+    relocate("org.yatopiamc.yatopia" to "org.bukkit.craftbukkit.libs") // Patina
+    relocate("me.titaniumtown" to "org.bukkit.craftbukkit.libs") // Patina
 }
 
 tasks.shadowJar {
