From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Thu, 1 Jul 2021 12:32:56 +0000
Subject: [PATCH] relocate


diff --git a/build.gradle.kts b/build.gradle.kts
index 5ecd6cf4d6f177781b1be5bf046c4a78b82efc43..8b80a5a54922b1b2c110cd66aef4105469e20d2b 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -123,6 +123,8 @@ relocation {
     relocate("org.bukkit.craftbukkit" to "org.bukkit.craftbukkit.v$craftbukkitPackageVersion") {
         exclude("org.bukkit.craftbukkit.Main*")
     }
+    relocate("org.yatopiamc.yatopia" to "org.bukkit.craftbukkit.libs") // Patina
+    relocate("me.titaniumtown" to "org.bukkit.craftbukkit.libs") // Patina
 }
 
 tasks.shadowJar {
