From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Fri, 23 Jun 2023 14:54:33 +0800
Subject: [PATCH] fix building issue


diff --git a/build.gradle.kts b/build.gradle.kts
index 5dd1d9c1ef5e83801d1982d1f51aef8625a83815..f56cf962a04503618c34f6c904600fbe0147d21b 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -49,7 +49,7 @@ dependencies {
 
     // Pufferfish start
     implementation("org.yaml:snakeyaml:1.32")
-    implementation ("me.carleslc.Simple-YAML:Simple-Yaml:1.8.4") {
+    implementation ("com.github.Carleslc:Simple-YAML:1.8.4") {
         exclude(group="org.yaml", module="snakeyaml")
     }
     // Pufferfish end
