From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Fri, 23 Jun 2023 14:57:20 +0800
Subject: [PATCH] remove broken import on purpur


diff --git a/src/main/java/org/purpurmc/purpur/PurpurConfig.java b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
index 98cc0b4bfe661879fd9b8f2eb7623e926ca82472..660aeeeb2caa8003550c8dca677a0e666baa91d3 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
@@ -22,8 +22,6 @@ import org.bukkit.command.Command;
 import org.bukkit.configuration.ConfigurationSection;
 import org.bukkit.configuration.InvalidConfigurationException;
 import org.bukkit.configuration.file.YamlConfiguration;
-import org.purpurmc.purpur.command.PurpurCommand;
-import org.purpurmc.purpur.task.TPSBarTask;
 
 import java.io.File;
 import java.io.IOException;
@@ -73,7 +71,7 @@ public class PurpurConfig {
         verbose = getBoolean("verbose", false);
 
         commands = new HashMap<>();
-        commands.put("purpur", new PurpurCommand("purpur"));
+        commands.put("purpur", new org.purpurmc.purpur.command.PurpurCommand("purpur"));
 
         version = getInt("config-version", 32);
         set("config-version", 32);
diff --git a/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java b/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
index b10d98af806def01e5c8e764162c1efaa5410f21..a3f9c922471ba5bfef56d2dbfbb093e8f8c0a949 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
@@ -12,11 +12,6 @@ import net.minecraft.world.level.Explosion;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.state.properties.Tilt;
-import org.purpurmc.purpur.entity.GlowSquidColor;
-import org.purpurmc.purpur.tool.Strippable;
-import org.purpurmc.purpur.tool.Tillable;
-import org.purpurmc.purpur.tool.Waxable;
-import org.purpurmc.purpur.tool.Weatherable;
 import org.apache.commons.lang.BooleanUtils;
 import org.bukkit.ChatColor;
 import org.bukkit.World;
