From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Thu, 1 Jul 2021 12:32:56 +0000
Subject: [PATCH] relocate


diff --git a/build.gradle.kts b/build.gradle.kts
index 0d935becfec698294851bc9ca0bcebf8aaeed9d7..8a63ca041614114b9d002e661e170d626c80c3c1 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -104,6 +104,8 @@ relocation {
     relocate("org.bukkit.craftbukkit" to "org.bukkit.craftbukkit.v$craftbukkitPackageVersion") {
         exclude("org.bukkit.craftbukkit.Main*")
     }
+    relocate("org.yatopiamc.yatopia" to "org.bukkit.craftbukkit.libs") // Patina
+    relocate("me.titaniumtown" to "org.bukkit.craftbukkit.libs") // Patina
 }
 
 tasks.shadowJar {
