From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Fri, 23 Jun 2023 14:57:20 +0800
Subject: [PATCH] remove broken import on purpur


diff --git a/src/main/java/org/purpurmc/purpur/PurpurConfig.java b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
index 28a65d2e40aee3f73a312785d396436df12cfcf2..2e1ad5aacbdf2211d2039889ee51480bdc3ee5fa 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
@@ -24,8 +24,6 @@ import org.bukkit.command.Command;
 import org.bukkit.configuration.ConfigurationSection;
 import org.bukkit.configuration.InvalidConfigurationException;
 import org.bukkit.configuration.file.YamlConfiguration;
-import org.purpurmc.purpur.command.PurpurCommand;
-import org.purpurmc.purpur.task.TPSBarTask;
 
 import java.io.File;
 import java.io.IOException;
@@ -75,7 +73,7 @@ public class PurpurConfig {
         verbose = getBoolean("verbose", false);
 
         commands = new HashMap<>();
-        commands.put("purpur", new PurpurCommand("purpur"));
+        commands.put("purpur", new org.purpurmc.purpur.command.PurpurCommand("purpur"));
 
         version = getInt("config-version", 35);
         set("config-version", 35);
diff --git a/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java b/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
index c13c05f7f971fc82b384bc2a9aa6c4c2841d6b84..9eca55959e200fe455f716d98d948e8f743c0fd2 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
@@ -12,11 +12,6 @@ import net.minecraft.world.item.Items;
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.state.properties.Tilt;
-import org.purpurmc.purpur.tool.Flattenable;
-import org.purpurmc.purpur.tool.Strippable;
-import org.purpurmc.purpur.tool.Tillable;
-import org.purpurmc.purpur.tool.Waxable;
-import org.purpurmc.purpur.tool.Weatherable;
 import org.apache.commons.lang.BooleanUtils;
 import org.bukkit.ChatColor;
 import org.bukkit.World;
