From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Fri, 23 Jun 2023 14:54:33 +0800
Subject: [PATCH] fix building issue


diff --git a/build.gradle.kts b/build.gradle.kts
index 381e1d611a663329cc7c7f446826f6a106d471e5..76804d663d7f42c6affe03c0c6cd9b0f66bd9b0c 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -53,7 +53,7 @@ dependencies {
 
     // Pufferfish start
     implementation("org.yaml:snakeyaml:1.32")
-    implementation ("me.carleslc.Simple-YAML:Simple-Yaml:1.8.4") {
+    implementation ("com.github.Carleslc:Simple-YAML:1.8.4") {
         exclude(group="org.yaml", module="snakeyaml")
     }
     // Pufferfish end
