From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Wed, 20 Jul 2022 18:11:32 +0800
Subject: [PATCH] Tweak default settings


diff --git a/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java b/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
index 38cb29c646ff496ffaa6553f98a565b71155c464..df1148b60a1a8758c9e69a63a28ebda82160300d 100644
--- a/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
+++ b/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
@@ -216,7 +216,7 @@ public class PufferfishConfig {
 	public static boolean enableAsyncEntityTracker;
 	public static boolean enableAsyncEntityTrackerInitialized;
 	private static void asyncEntityTracker() {
-		boolean temp = getBoolean("enable-async-entity-tracker", false,
+		boolean temp = getBoolean("enable-async-entity-tracker", true, // Patina - defaulted to true
 				"Whether or not async entity tracking should be enabled.");
 		if (!enableAsyncEntityTrackerInitialized) {
 			enableAsyncEntityTrackerInitialized = true;
@@ -227,7 +227,7 @@ public class PufferfishConfig {
 	public static boolean enableAsyncPathfinding;
 	public static boolean enableAsyncPathfindingInitialized;
 	private static void asyncPathfinding() {
-		boolean temp = getBoolean("enable-async-pathfinding", false,
+		boolean temp = getBoolean("enable-async-pathfinding", true, // Patina - defaulted to true
 				"Whether or not async pathfinding should be enabled.");
 		if (!enableAsyncPathfindingInitialized) {
 			enableAsyncPathfindingInitialized = true;
