From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Thu, 1 Jul 2021 12:11:01 +0000
Subject: [PATCH] Remove unnecessary `LogManager.getLogger()`


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index b998df05702e2aa7adbd70cb3f101b70015a78cb..5b25d54661058c7728e8c41bf5d67893ab610f1b 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -163,7 +163,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
         @Override
         public synchronized void setSeed(long seed) {
             if (locked) {
-                LogManager.getLogger().error("Ignoring setSeed on Entity.SHARED_RANDOM", new Throwable());
+                LOGGER.error("Ignoring setSeed on Entity.SHARED_RANDOM", new Throwable()); // Patina - LOGGER
             } else {
                 super.setSeed(seed);
                 locked = true;
@@ -2323,7 +2323,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
                 try {
                     spawnReason = org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.valueOf(spawnReasonName);
                 } catch (Exception ignored) {
-                    LogManager.getLogger().error("Unknown SpawnReason " + spawnReasonName + " for " + this);
+                    LOGGER.error("Unknown SpawnReason " + spawnReasonName + " for " + this); // Patina - LOGGER
                 }
             }
             if (spawnReason == null) {
