From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Fri, 23 Jun 2023 15:07:32 +0800
Subject: [PATCH] fix decompile


diff --git a/src/main/java/net/minecraft/world/level/block/SignBlock.java b/src/main/java/net/minecraft/world/level/block/SignBlock.java
index 91bb294be2cd8ab3467a62006a5a2751e0bec4ba..237b0317660dff46ff2b88c3ad757389415c0e43 100644
--- a/src/main/java/net/minecraft/world/level/block/SignBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/SignBlock.java
@@ -76,9 +76,9 @@ public abstract class SignBlock extends BaseEntityBlock implements SimpleWaterlo
     public InteractionResult use(BlockState state, Level world, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit) {
         ItemStack itemStack = player.getItemInHand(hand);
         Item item = itemStack.getItem();
-        Item signBlockEntity = itemStack.getItem();
+        Item signBlockEntity0 = itemStack.getItem(); // fix decompile
         SignApplicator var10000;
-        if (signBlockEntity instanceof SignApplicator signApplicator) {
+        if (signBlockEntity0 instanceof SignApplicator signApplicator) { // fix decompile
             var10000 = signApplicator;
         } else {
             var10000 = null;
@@ -86,8 +86,8 @@ public abstract class SignBlock extends BaseEntityBlock implements SimpleWaterlo
 
         SignApplicator signApplicator2 = var10000;
         boolean bl = signApplicator2 != null && player.mayBuild();
-        BlockEntity bl2 = world.getBlockEntity(pos);
-        if (bl2 instanceof SignBlockEntity signBlockEntity) {
+        BlockEntity bl20 = world.getBlockEntity(pos); // fix decompile
+        if (bl20 instanceof SignBlockEntity signBlockEntity) { // fix decompile
             if (!world.isClientSide) {
                 boolean bl2 = signBlockEntity.isFacingFrontText(player);
                 SignText signText = signBlockEntity.getText(bl2);
