From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Thu, 1 Jul 2021 12:32:56 +0000
Subject: [PATCH] relocate


diff --git a/build.gradle.kts b/build.gradle.kts
index c6e723b09fa962d037ec6268dd4ec5f74a4c08cf..a31507ea4eabb2b026264cefac1bc810260d588c 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -109,6 +109,8 @@ relocation {
     relocate("org.bukkit.craftbukkit" to "org.bukkit.craftbukkit.v$packageVersion") {
         exclude("org.bukkit.craftbukkit.Main*")
     }
+    relocate("org.yatopiamc.yatopia" to "org.bukkit.craftbukkit.libs") // Patina
+    relocate("me.titaniumtown" to "org.bukkit.craftbukkit.libs") // Patina
 }
 
 tasks.shadowJar {
