From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Thu, 1 Jul 2021 12:32:56 +0000
Subject: [PATCH] relocate


diff --git a/build.gradle.kts b/build.gradle.kts
index 6700e1b580d60483c1e534d3241edd33b102574f..b8b856d194076f8e166dac1bf6c7270926bea6f7 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -100,6 +100,8 @@ relocation {
     relocate("org.bukkit.craftbukkit" to "org.bukkit.craftbukkit.v$packageVersion") {
         exclude("org.bukkit.craftbukkit.Main*")
     }
+    relocate("org.yatopiamc.yatopia" to "org.bukkit.craftbukkit.libs") // Patina
+    relocate("me.titaniumtown" to "org.bukkit.craftbukkit.libs") // Patina
 }
 
 tasks.shadowJar {
