From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Tue, 15 Feb 2022 12:48:21 +0000
Subject: [PATCH] Update JettPack's redstone patch


diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
index 5525d3280b6e838ae7bc474a58f3200067b0e9d0..bfcb920175c198cc39827213139e4ffaab7a5319 100644
--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
@@ -71,8 +71,8 @@ public class PaperConfig {
         commands.put("paper", new PaperCommand("paper"));
         commands.put("mspt", new MSPTCommand("mspt"));
 
-        version = getInt("config-version", 25);
-        set("config-version", 25);
+        version = getInt("config-version", 24);
+        set("config-version", 24);
         readConfig(PaperConfig.class, null);
     }
 
diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
index 14c21ab46cb5e1260015f4fda52d0cb566d2cee8..1c3a54bf0fe9551d3ed6bff03b68d468db5d2031 100644
--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
@@ -83,14 +83,8 @@ public class PaperWorldConfig {
 
     public RedstoneAlgo redstoneAlgo = RedstoneAlgo.EIGENCRAFT;
     private void redstoneAlgo() {
-        String redstoneAlgoString = getString("redstone-algo", "eigencraft").toLowerCase().trim();
-        if (PaperConfig.version < 25) {
-            boolean oldUseEigenCraft = getBoolean("use-faster-eigencraft-redstone", false);
-            if (!oldUseEigenCraft) {
-                redstoneAlgoString = "vanilla";
-                set("redstone-algo", "vanilla");
-            }
-        }
+        String redstoneAlgoString = getString("redstone-algo", getBoolean("use-faster-eigencraft-redstone", false) ? "eigencraft" : "vanilla").toLowerCase().trim(); // Patina
+        set("redstone-algo", redstoneAlgoString); // Patina
         switch (redstoneAlgoString) {
             case "vanilla":
                 log("Using vanilla redstone algorithm.");
@@ -105,8 +99,8 @@ public class PaperWorldConfig {
                 redstoneAlgo = RedstoneAlgo.ALTERNATE_CURRENT;
                 break;
             default:
-                logError("Warning: redstone-algo set to an invalid value of " + redstoneAlgoString + " must be one of: vanilla, eigencraft, alternate-current. Defaulting to eigencraft.");
-                redstoneAlgo = RedstoneAlgo.EIGENCRAFT;
+                logError("Warning: redstone-algo set to an invalid value of " + redstoneAlgoString + " must be one of: vanilla, eigencraft, alternate-current. Defaulting to vanilla."); // Patina
+                redstoneAlgo = RedstoneAlgo.VANILLA; // Patina
                 break;
         }
     }
