From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Mon, 2 Sep 2024 16:22:46 +1200
Subject: [PATCH] remove getServerName from purpur


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 5f5f513e538df378884cda2da13988d96bf24e63..f90c7414ac4dc7a738f97b73ec8516604348fc19 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -3230,10 +3230,6 @@ public final class CraftServer implements Server {
         return this.potionBrewer;
     }
     // Paper end
-    public String getServerName() {
-        return this.getProperties().serverName;
-    }
-
     @Override
     public boolean isLagging() {
         return getServer().lagging;
