From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Mon, 2 Sep 2024 16:22:46 +1200
Subject: [PATCH] remove getServerName from purpur


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 3c21bde0724e325a1db13f8f02142b2b7372fab1..f5c3a5a8dc0ae2cff8d534a6057e9bb315636cf9 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -3233,10 +3233,6 @@ public final class CraftServer implements Server {
         return this.potionBrewer;
     }
     // Paper end
-    public String getServerName() {
-        return this.getProperties().serverName;
-    }
-
     @Override
     public boolean isLagging() {
         return getServer().lagging;
