From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Fri, 6 Oct 2023 22:45:46 +0800
Subject: [PATCH] fix rebuild issue (temporarily)


diff --git a/build.gradle.kts b/build.gradle.kts
index 57beb8a1c12e4935c37363bf62bd7ed6f45e829a..d87904ebc02dbc817d61224ead776b8cfedf2829 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -227,7 +227,5 @@ val runtimeClasspathForRunDev = sourceSets.main.flatMap { src ->
 }
 tasks.registerRunTask("runDev") {
     description = "Spin up a non-relocated Mojang-mapped test server"
-    classpath(tasks.filterProjectDir.flatMap { it.outputJar })
-    classpath(runtimeClasspathForRunDev)
-    jvmArgs("-DPaper.isRunDev=true")
+    classpath(sourceSets.main.map { it.runtimeClasspath })
 }
