From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Fri, 6 Oct 2023 22:45:46 +0800
Subject: [PATCH] fix rebuild issue (temporarily)


diff --git a/build.gradle.kts b/build.gradle.kts
index c8ec0238c0c29905f08865cfbbf7b3aff9fa1d20..7e9273f8274e50d72d84866289e72363e3f305f1 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -229,7 +229,5 @@ val runtimeClasspathForRunDev = sourceSets.main.flatMap { src ->
 }
 tasks.registerRunTask("runDev") {
     description = "Spin up a non-relocated Mojang-mapped test server"
-    classpath(tasks.filterProjectDir.flatMap { it.outputJar })
-    classpath(runtimeClasspathForRunDev)
-    jvmArgs("-DPaper.isRunDev=true")
+    classpath(sourceSets.main.map { it.runtimeClasspath })
 }
