From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Fri, 6 Oct 2023 22:45:46 +0800
Subject: [PATCH] fix rebuild issue (temporarily)


diff --git a/build.gradle.kts b/build.gradle.kts
index 43414c00475c3273e1ce76986b89acbfba8d4e53..d6767d64931db9d99bd56c349b1c3976abecbf6a 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -228,7 +228,5 @@ val runtimeClasspathForRunDev = sourceSets.main.flatMap { src ->
 }
 tasks.registerRunTask("runDev") {
     description = "Spin up a non-relocated Mojang-mapped test server"
-    classpath(tasks.filterProjectDir.flatMap { it.outputJar })
-    classpath(runtimeClasspathForRunDev)
-    jvmArgs("-DPaper.isRunDev=true")
+    classpath(sourceSets.main.map { it.runtimeClasspath })
 }
