From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Fri, 11 Jun 2021 12:01:35 +0000
Subject: [PATCH] fixup EMC


diff --git a/src/main/java/net/minecraft/world/level/block/entity/TileEntityLectern.java b/src/main/java/net/minecraft/world/level/block/entity/TileEntityLectern.java
index f09457ce1e871df6ea49c6bc08ea5c178301fdb6..31b03823fcfc33b8b04eccd5ad1fdcef4c33968e 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/TileEntityLectern.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/TileEntityLectern.java
@@ -44,12 +44,6 @@ public class TileEntityLectern extends TileEntity implements Clearable, ITileInv
     // CraftBukkit start - add fields and methods
     public final IInventory inventory = new LecternInventory();
     public class LecternInventory implements IInventory {
-        // EMC start
-        public TileEntityLectern getLectern() {
-            return TileEntityLectern.this;
-        }
-        // EMC end
-
         public List<HumanEntity> transaction = new ArrayList<>();
         private int maxStack = 1;
 
@@ -88,6 +82,7 @@ public class TileEntityLectern extends TileEntity implements Clearable, ITileInv
             return (Lectern) TileEntityLectern.this.getOwner();
         }
 
+        // Patina - it is here
         public TileEntityLectern getLectern() {
             return TileEntityLectern.this;
         }
