From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Thu, 1 Jul 2021 12:32:56 +0000
Subject: [PATCH] relocate


diff --git a/build.gradle.kts b/build.gradle.kts
index ba4c4f37efe1be04ae41faf2861a3ca1808cdbaa..8dceeda57022d45b02ad766a24d6756c6aa0ad7d 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -121,6 +121,7 @@ relocation {
     relocate("org.bukkit.craftbukkit" to "org.bukkit.craftbukkit.v$packageVersion") {
         exclude("org.bukkit.craftbukkit.Main*")
     }
+    relocate("org.yatopiamc.yatopia.server" to "org.bukkit.craftbukkit.libs") // Patina
 
     fun cb(pack: String) = "org.bukkit.craftbukkit.libs.$pack"
 
