From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Thu, 1 Jul 2021 12:32:56 +0000
Subject: [PATCH] relocate


diff --git a/build.gradle.kts b/build.gradle.kts
index 3fcd028b8aa2dd330afb3075ea0680847416c79e..809e02e1f9fc63854b961e06588bce12b8e9280d 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -120,6 +120,7 @@ relocation {
     relocate("org.bukkit.craftbukkit" to "org.bukkit.craftbukkit.v$packageVersion") {
         exclude("org.bukkit.craftbukkit.Main*")
     }
+    relocate("org.yatopiamc.yatopia.server" to "org.bukkit.craftbukkit.libs") // Patina
 
     fun cb(pack: String) = "org.bukkit.craftbukkit.libs.$pack"
 
