From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Thu, 1 Jul 2021 12:32:56 +0000
Subject: [PATCH] relocate


diff --git a/build.gradle.kts b/build.gradle.kts
index 9936f1d7badb6cf666e7aae454c88836ab01a8a7..9510537b51ebe5236e07ad74a6f0cee445d8d68b 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -116,6 +116,7 @@ relocation {
     relocate("org.bukkit.craftbukkit" to "org.bukkit.craftbukkit.v$packageVersion") {
         exclude("org.bukkit.craftbukkit.Main*")
     }
+    relocate("org.yatopiamc.yatopia.server" to "org.bukkit.craftbukkit.libs") // Patina
 
     fun cb(pack: String) = "org.bukkit.craftbukkit.libs.$pack"
 
