From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Thu, 1 Jul 2021 12:32:56 +0000
Subject: [PATCH] relocate


diff --git a/build.gradle.kts b/build.gradle.kts
index de4a72e9d9893d16f5fc130bb40ad7197f14e880..6a5b256154ec09a5048d90392c6f5d929425ad60 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -147,6 +147,7 @@ tasks.shadowJar {
     relocate("org.bukkit.craftbukkit", "org.bukkit.craftbukkit.v$packageVersion") {
         exclude("org.bukkit.craftbukkit.Main*")
     }
+    relocate("org.yatopiamc.yatopia.server", "org.bukkit.craftbukkit.libs") // Patina
     listOf(
         "jline", "org.apache.commons.codec", "org.apache.commons.io", // Paper - don't relocate fastutil
         "org.apache.commons.lang3", "org.apache.http", "org.apache.maven", "org.codehaus.plexus",
