From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Thu, 1 Jul 2021 12:32:56 +0000
Subject: [PATCH] relocate


diff --git a/build.gradle.kts b/build.gradle.kts
index 47b5412252b5765411a565e8b2eeeec8f2c89c6e..0fa23bb0fc00e19f6d498b8f379525094be5d658 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -129,6 +129,7 @@ tasks.shadowJar {
     relocate("org.bukkit.craftbukkit", "org.bukkit.craftbukkit.v$packageVersion") {
         exclude("org.bukkit.craftbukkit.Main*")
     }
+    relocate("org.yatopiamc.yatopia.server", "org.bukkit.craftbukkit.libs") // Patina
     listOf(
         "jline", "org.apache.commons.codec", "org.apache.commons.io", // Paper - don't relocate fastutil
         "org.apache.commons.lang3", "org.apache.http", "org.apache.maven", "org.codehaus.plexus",
